logstash-core/lib/logstash/compiler/lscl.rb

```diff
@@ -115,7 +115,7 @@ def expr_attributes
         elsif existing.kind_of?(::Array)
           hash[k] = existing.push(*v)
         else
-          hash[k] = existing + v
+          hash[k] = [existing, v] unless v == existing
```

Comment:
I personally would prefer that we fail, indicating that directive `k` was provided twice and we don't know how to merge the two. 

Silently ignoring one instance of the directive IFF the two are equal (and introducing a change in behaviour by emitting an array if they're not) feels like it's going to bite someone, and I would rather bite them with an explicit call-out of what is ambiguous and needs to get fixed than with an arbitrary change in unspecified behaviour.

Reply:
To clarify, the behavior above (multiple scalar values merged into an array so longer as they don't duplicate existing values) exists already in the Ruby execution engine through a later invocation of [`hash_merge_many`](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/util.rb#L131) on the config values. This change aligns the behavior of the Java execution engine with that in code that is common to both of them. @jsvd, @yaauie, under those circumstances, do you think warnings or errors should be produced?

Reply:
Good question. We can, for now, align the behaviour in a bwc way (essentially what this PR does), and open a separate discussion about gradually preventing double options.

Can you also update the description to clarify this happens only in the java exec?

Reply:
After digging in to the linked code, I agree that this pulls Java LSCL into alignment with Ruby Grammar implementation.